### PR TITLE
Docs: Terrajet => Upjet

### PR DESCRIPTION
**Pull request by luebken**
Signed-off-by: Matthias Luebken <matthias.luebken@gmail.com>

### Description of your changes

Referencing Upjet instead of Terrajet to generate providers.

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by ytsarev — APPROVED** [review]
LGTM

File: docs/contributing/provider_development_guide.md
Comment (hasheddan) [inline docs/contributing/provider_development_guide.md:655]: Appears this link is broken:
```suggestion
[Upjet]: https://github.com/upbound/upjet
```

**Review by hasheddan — DISMISSED** [review]

File: docs/contributing/provider_development_guide.md
Comment (negz) [reply docs/contributing/provider_development_guide.md:655]: Ping @luebken. This needs an update before we can merge.

**Comment by negz** [comment]
@luebken Looks like your latest commit is missing DCO sign-off. Could you please sign it.

**Review by negz — APPROVED** [review]

**Review by hasheddan — APPROVED** [review]
LGTM, though it looks like DCO is still failing.

**Pull request merged**
